File: kangaroo_app_sdk_ios/ios/Classes/features/strings/StringsHandler.swift
import Foundation
import Flutter
import KangarooAppSDKiOS

class StringsHandler: NSObject, FlutterStreamHandler, PluginChannelHandler {
    var sink: FlutterEventSink?

    var methodChannel: String = "customer_sdk/methods/get_strings"

    var eventChannel: String = "customer_sdk/events/get_strings"

    func onMethodCall(call: FlutterMethodCall) -> Void? {
        StringsHandler.getStrings(call: call)
    }

    func getStreamHandler() -> (FlutterStreamHandler & NSObjectProtocol)? {
        return self
    }


    static func getStrings(call: FlutterMethodCall) {
        StringsApi().getStrings()

        
    }

    func onListen(withArguments arguments: Any?, eventSink events: @escaping
        FlutterEventSink) -> FlutterError? {
        sink = events
        StringsApi().stringsState
            .serializeStringsState().observe { result in
            switch result {
            case let result as SerializedResultIdle:
                self.sink?(result.state)
            case let result as SerializedResultLoading:
                self.sink?(result.state)
            case let result as SerializedResultSuccess:
                self.sink?(result.data)
            case let result as SerializedResultUnauthorizedError:
                self.sink?(result.error)
            case let result as SerializedResultUnknownError:
                self.sink?(result.error)
            default:
                break
            }
        }
        return nil
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}

File: kangaroo_app_sdk_android/android/src/main/kotlin/com/kangaroo/flutterplugin/android/features/strings/StringsHandler.kt
package com.kangaroo.flutterplugin.android.features.strings

import com.kangaroo.flutterplugin.android.base.PluginChannelHandler
import com.kangaroo.flutterplugin.android.base.pushSerializedResultToEventSink
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodCall
import kangaroorewards.appsdk.features.strings.StringsApi
import kangaroorewards.appsdk.features.strings.serializeStringsState


class StringsHandler : EventChannel.StreamHandler, PluginChannelHandler {
    var sink: EventChannel.EventSink? = null

    override val methodChannel: String
        get() = "customer_sdk/methods/get_strings"

    override val eventChannel: String
        get() = "customer_sdk/events/get_strings"

    override fun onMethodCall(call: MethodCall): Unit? {
        return getStrings(call)
    }

    override fun getStreamHandler(): EventChannel.StreamHandler {
        return this
    }

    companion object {
        fun getStrings(call: MethodCall): Unit? {
            StringsApi().getStrings()
            return null
        }
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
        StringsApi().stringsState.serializeStringsState().observe { result ->
            pushSerializedResultToEventSink<StringsApi>(sink, result)
        }
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }
}
